services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/QinQFlowSpec.groovy

```diff
@@ -0,0 +1,365 @@
+        }
+        when: "Try to create a QinQ flow when at least on switch doesn't support multi table mode"
+
+        def flow = flowHelperV2.randomFlow(swP)
+        flow.source.innerVlanId = 4093
+        flow.destination.innerVlanId = 3904
+        northboundV2.addFlow(flow)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Could not create flow"
+
+        cleanup: "Revert system to original state"
+        northbound.updateSwitchProperties(swP.src.dpId, initSrcSwProps)
+        Wrappers.wait(RULES_INSTALLATION_TIME, 1) {
+            assert northbound.getSwitchRules(swP.src.dpId).flowEntries*.cookie.sort() == swP.src.defaultCookies.sort()
+        }
+    }
+
+    @Ignore("is not implemented yet")
+    @Unroll
+    def "System doesn't allow to create a QinQ flow with incorrect innerVlanIds\
+(src:srcInnerVlanId, dst:dstInnerVlanId)"() {
+        when: "Try to create a QinQ flow with incorrect innerVlanId"
+        def swP = topologyHelper.getAllNeighboringSwitchPairs().find {
+            [it.src, it.dst].any { northbound.getSwitchProperties(it.dpId).multiTable }
+        } ?: assumeTrue("No suiting switches found", false)
+
+        def flow = flowHelperV2.randomFlow(swP)
+        flow.source.innerVlanId = srcInnerVlanId
+        flow.destination.innerVlanId = dstInnerVlanId
+        northboundV2.addFlow(flow)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        //TODO(andriidovhan) add error message
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Could not create flow"
+
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+
+        where:
+        srcInnerVlanId | dstInnerVlanId | errorMsg
+        4096           | 10             | ""
+        10             | -1             | ""
+    }
+
+    @Ignore("is not implemented yet")
+    def "System doesn't allow to create/update/delete a QinQ flow via APIv1"() {
+        given: "Two switches with enabled multi table mode"
+        def swP = topologyHelper.getAllNeighboringSwitchPairs().find {
+            [it.src, it.dst].every { northbound.getSwitchProperties(it.dpId).multiTable }
+        } ?: assumeTrue("No suiting switches found", false)
+
+        when: "Create a QinQ flow via APIv1"
+        def flowV1 = flowHelper.randomFlow(swP)
+        flowV1.source.innerVlanId = 100
+        flowV1.destination.innerVlanId = 200
+        northbound.addFlow(flowV1)
+
+        then: "Human readable error is returned"
+        def eCreate = thrown(HttpClientErrorException)
+        eCreate.rawStatusCode == 400
+        //TODO(andriidovhan) add error message
+        eCreate.responseBodyAsString.to(MessageError).errorMessage == "Could not create the flow"
+
```

Comment:
new test starts here

Reply:
initially it was two separate tests, but then I had second thoughts)

fixed